legacy/src/classes/models/User.php

```diff
@@ -2195,14 +2195,16 @@ public function addUserRole($user_id, $role, $room_id)
     $user_id = $this->converters->checkUserId($user_id);
     $room_id = $this->converters->checkRoomId($room_id);
 
+    $this->db->beginTransaction();
     $stmt = $this->db->query('SELECT hash_id FROM ' . $this->db->au_rooms . ' WHERE id = :room_id');
     $this->db->bind(':room_id', $room_id);
     $room_hash = $this->db->resultSet()[0]["hash_id"];
 
```

Comment:
as far as i understand, there's no need for the first query to be a part of the transaction?
```suggestion
    $stmt = $this->db->query('SELECT hash_id FROM ' . $this->db->au_rooms . ' WHERE id = :room_id');
    $this->db->bind(':room_id', $room_id);
    $room_hash = $this->db->resultSet()[0]["hash_id"];

    $this->db->beginTransaction();
```